code/game/machinery/doors/airlock.dm

```diff
@@ -1037,12 +1037,14 @@
 	else if(locked)
 		to_chat(user, "<span class='warning'>The airlock's bolts prevent it from being forced!</span>")
 	else if( !welded && !operating)
-		if(!beingcrowbarred) //being fireaxe'd
-			var/obj/item/twohanded/fireaxe/F = I
-			if(F.wielded)
-				INVOKE_ASYNC(src, (density ? .proc/open : .proc/close), 2)
-			else
+		if(!beingcrowbarred) //being fireaxe'd (maybe)
+			var/obj/item/twohanded/fireaxe/F = null
```

Comment:
you added an istype so why not ONLY create the variable when you know the object is a fireaxe? Then you can also avoid the `if(F)` check on 1044

Reply:
Far as I can picture it, if I do that, I'll need an extra `else <cycle airlock code>` inside that if statement too. As is, since we only need to stop if the tool is both a fireaxe and also not wielded, I can combine both checks together into one if statement.

I might be overthinking it, though. I'll look into it later on today.

Reply:
nani?

this is cobbydm do not 1:1 copy
```DM
if(istype(I,fireaxe/type))
    var/type/stuff/here/fireaxe/F = I
    if(!F.wielded)
        to_chat(user,msg)
    else
        INVOKE_ASYNC(stuff)

Reply:
```
if(!beingcrowbarred)
    if(istype(I,fireaxe/type))
        var/type/stuff/here/fireaxe/F = I
        if(!F.wielded)
            to_chat(user,msg)
        else
            INVOKE_ASYNC(stuff) //if it's not a crowbar but is a fireaxe and wielded
    else
        INVOKE_ASYNC(stuff) //if it's not a crowbar or a fireaxe but called this proc anyway
else
    INVOKE_ASYNC(stuff) //if it's a crowbar
```

Reply:
I'm gonna mark this as resolved, since I re-wrote the code. Anyway, the variable creation did get moved to after the `if`.